code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -83,17 +87,42 @@
 					visible_message("<span class='warning'>[src] chews through the [C].</span>")
 	for(var/obj/item/reagent_containers/food/snacks/cheesewedge/cheese in range(1, src))
 		if(prob(10))
-			var/cap = CONFIG_GET(number/ratcap)
-			if(LAZYLEN(SSmobs.cheeserats) >= cap)
-				visible_message("<span class='warning'>[src] carefully eats the cheese, hiding it from the [cap] mice on the station!</span>")
-				qdel(cheese)
-				return
-			var/mob/living/newmouse = new /mob/living/simple_animal/mouse(loc)
-			SSmobs.cheeserats += newmouse
-			visible_message("<span class='notice'>[src] nibbles through the [cheese], attracting another mouse!</span>")
+			be_fruitful()
 			qdel(cheese)
 			return
 
+/mob/living/simple_animal/mouse/Life(seconds, times_fired)
+	. = ..()
+	if(stat != CONSCIOUS)
+		return
+	for(var/obj/item/reagent_containers/food/snacks/royalcheese/bigcheese in range(1, src))
+		qdel(bigcheese)
+		evolve()
+		return
```

Comment:
Could you handle this in `handle_automated_action()`
An attack action for a mouse player would work as well.
No need to pollute `Life()` anymore in my opinion.